lib/elixir/lib/kernel.ex

```diff
@@ -912,6 +912,11 @@ defmodule Kernel do
   @doc """
   Concatenates two lists.
 
+  The complexity of `a ++ b` is proportional to `length(a)`, so avoid appending
+  to lists of arbitrary length ie: `list ++ [item]`
```

Comment:
I added an example of what to do so it was a clearer contrast to the recommended course of prepending.